src/relayFeeCalculator/chain-queries/baseQuery.ts

```diff
@@ -61,17 +62,25 @@ export class QueryBase implements QueryInterface {
    * Retrieves the current gas costs of performing a fillRelay contract at the referenced SpokePool.
    * @param deposit V3 deposit instance.
    * @param relayerAddress Relayer address to simulate with.
-   * @param gasPrice Optional gas price to use for the simulation.
-   * @param gasUnits Optional gas units to use for the simulation.
+   * @param options
+   * @param options.gasPrice Optional gas price to use for the simulation.
+   * @param options.gasUnits Optional gas units to use for the simulation.
+   * @param options.omitMarkup Optional flag to omit the gas markup.
+   * @param options.transport Optional transport object for custom gas price retrieval.
    * @returns The gas estimate for this function call (multiplied with the optional buffer).
    */
   async getGasCosts(
     deposit: Deposit,
     relayer = DEFAULT_SIMULATED_RELAYER_ADDRESS,
-    gasPrice = this.fixedGasPrice,
-    gasUnits?: BigNumberish,
-    omitMarkup?: boolean
+    options: Partial<{
+      gasPrice: BigNumberish;
+      gasUnits: BigNumberish;
+      omitMarkup: boolean;
+      transport: Transport;
+    }> = {}
```

Comment:
+1 for this approach - I really think it improves usability to eliminate the need for long convoluted sets of parameters.